### ISSUE TITLE
[BUG] - Using Dropdown component in Next.JS throws a runtime exception

### ISSUE THREAD

**Issue by iyxan23** [heroui-inc/heroui#1403]
### NextUI Version

^2.0.15

### Describe the bug

Using the Dropdown component as stated in the docs causes a Runtime Exception when the trigger is clicked. The error itself does not provide any meaningful information for debugging: `Error: Unknown element <[object Object]> in collection.`, so I believe this is an internal bug. Source: https://nextui.org/docs/components/dropdown

### Steps to Reproduce the Bug or Issue

1. Install NextUI as described in the docs
2. Copy the code from https://nextui.org/docs/components/dropdown#usage
3. Clicks the dropdown button
4. Raised a cryptic runtime exception `Error: Unknown element <[object Object]> in collection.`

Additional info: I'm using Vivaldi, a derivative of chrome.

### Expected behavior

I expected the dropdown to show and to not raise an exception.

### Screenshots or Videos

https://github.com/nextui-org/nextui/assets/31884435/455677d5-ee0f-45f6-bf88-97660457b20a

### Operating System Version

Fedora 38 KDE Plasma; Linux laptop 6.3.11-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Jul  2 13:17:31 UTC 2023 x86_64 GNU/Linux

### Browser

Other (add additonal context)

**Comment by robodove** [comment]
Happens on Tabs too @tabs/2.0.11

For now wrapping them in a client component ("use client") makes them work.

**Comment by jrgarciadev** [comment]
Hey @iyxan23 we are waiting for the next [react-aria](https://github.com/adobe/react-spectrum) release to get this issue fixed, the current collection approach is compiled by next.js in a different way so it removes the `use client;` directive, in the meantime please add the directive on top of your files for collection based components -> Table, Accordion, Dropdown, and Tabs.

**Comment by iyxan23** [comment]
> Hey @iyxan23 we are waiting for the next [react-aria](https://github.com/adobe/react-spectrum) release to get this issue fixed, the current collection approach is compiled by next.js in a different way so it removes the `use client;` directive, in the meantime please add the directive on top of your files for collection based components -> Table, Accordion, Dropdown, and Tabs.

Alright! I'll be keeping an eye on new updates 😉
Thanks for the quick response

**Comment by guivalerioS** [comment]
Hy guys, Do we have an estimate of when this will be fixed?

**Comment by JimmyRice** [comment]
The Select component doesn't seem to be working either.

**Comment by CitizenBeta** [comment]
> Hey @iyxan23 we are waiting for the next [react-aria](https://github.com/adobe/react-spectrum) release to get this issue fixed, the current collection approach is compiled by next.js in a different way so it removes the `use client;` directive, in the meantime please add the directive on top of your files for collection based components -> Table, Accordion, Dropdown, and Tabs.

What exactly do we add to our code to temporarily resolve it? I'm running into this issue with dropdowns.

**Comment by CitizenBeta** [comment]
If anyone else comes here, this is what it's supposed to look like, and it does fix this error perfectly:

![Screenshot from 2023-09-27 17-03-02](https://github.com/nextui-org/nextui/assets/595049/b44266da-ae29-4aa6-857e-94c78f994e01)

**Comment by robodove** [comment]
> If anyone else comes here, this is what it's supposed to look like, and it does fix this error perfectly:
> 
> 
> 
> ![Screenshot from 2023-09-27 17-03-02](https://github.com/nextui-org/nextui/assets/595049/b44266da-ae29-4aa6-857e-94c78f994e01)
> 
> 

No need to mark the whole page or larger server components as client components because of this.

You can just do this

```
"use client"
export { Select } from "@nextui-org/select"
```

Now you can just import the component from this file and you're done

**Comment by mohammad1381** [comment]
> > If anyone else comes here, this is what it's supposed to look like, and it does fix this error perfectly:
> > ![Screenshot from 2023-09-27 17-03-02](https://github.com/nextui-org/nextui/assets/595049/b44266da-ae29-4aa6-857e-94c78f994e01)
> 
> No need to mark the whole page or larger server components as client components because of this.
> 
> You can just do this
> 
> ```
> "use client"
> export { Select } from "@nextui-org/select"
> ```
> 
> Now you can just import the component from this file and you're done

For some them not working this way. For example it works for `Button `but not work for `Tabs`

**Comment by sspaeth-r7** [comment]
Is there any way to work around this on a component which must be server-side generated?

**Comment by yahorbarkouski** [comment]
Hey there, any updates for tabs?

**Comment by YTG2G3** [comment]
Still doesn't seem to be working. The workaround is creating a mess between client component and server action... I hope this gets fixed soon.

**Comment by DylanAlloy** [comment]
If you put your offending Accordion code into a new component, put `"use script";` into that component, and import it into your server-borne .jsx file (being sure to remove `"use script";` from there if you put it as a temp fix), it will work.

**Comment by caanyp24** [comment]
same issue here any updates?

**Comment by ahomu** [comment]
same issue...

**Comment by ArianSha** [comment]
+1

**Comment by elanclarkson** [comment]
+1

**Comment by ygd** [comment]
+1

**Comment by usuarez** [comment]
I fix this removing the NextUIProvider from the `layout.tsx` of your nextjs application in `src/app/layout.tsx` (i put providers there cuz i see it like the root app component) you can set it as a client component or don't touch.
Create a new layout component in other directory and import providers there, it works perfectly for me

**Comment by vvchldmsdn** [comment]
> I fix this removing the NextUIProvider from the `layout.tsx` of your nextjs application in `src/app/layout.tsx` (i put providers there cuz i see it like the root app component). Create a new layout component in other directory and import providers there, it works perfectly for me

You meant "other directory" insidie app directory like 'src/app/something/layout' or 'src/something/layout' ? or either works. Thanks in advance

**Comment by usuarez** [comment]
> > I fix this removing the NextUIProvider from the `layout.tsx` of your nextjs application in `src/app/layout.tsx` (i put providers there cuz i see it like the root app component). Create a new layout component in other directory and import providers there, it works perfectly for me
> 
> You meant "other directory" insidie app directory like 'src/app/something/layout' or 'src/something/layout' ? or either works. Thanks in advance

Like 'src/app/something/layout' or 'src/something/layout' works perfect, but the best way is using the layout.tsx. I found a fix for layout.tsx, let me make some tests before post the fix

**Comment by usuarez** [comment]
> > I fix this removing the NextUIProvider from the `layout.tsx` of your nextjs application in `src/app/layout.tsx` (i put providers there cuz i see it like the root app component). Create a new layout component in other directory and import providers there, it works perfectly for me
> 
> You meant "other directory" insidie app directory like 'src/app/something/layout' or 'src/something/layout' ? or either works. Thanks in advance

Hello i found a fix for that:
1. we don't need import the rootLayout in our pages (my first mistake)
2. add the 'use client' directive is mandatory for our client side components, in this case if you import nextUI components in the page.tsx then need to declare 'use client' at the top of the file.

**Comment by vvchldmsdn** [comment]
> > > I fix this removing the NextUIProvider from the `layout.tsx` of your nextjs application in `src/app/layout.tsx` (i put providers there cuz i see it like the root app component). Create a new layout component in other directory and import providers there, it works perfectly for me
> > 
> > 
> > You meant "other directory" insidie app directory like 'src/app/something/layout' or 'src/something/layout' ? or either works. Thanks in advance
> 
> Hello i found a fix for that:
> 
> 1. we don't need import the rootLayout in our pages (my first mistake)
> 2. add the 'use client' directive is mandatory for our client side components, in this case if you import nextUI components in the page.tsx then need to declare 'use client' at the top of the file.

I love you

**Comment by robertgoacher** [comment]
> Hey @iyxan23 we are waiting for the next [react-aria](https://github.com/adobe/react-spectrum) release to get this issue fixed, the current collection approach is compiled by next.js in a different way so it removes the `use client;` directive, in the meantime please add the directive on top of your files for collection based components -> Table, Accordion, Dropdown, and Tabs.

Any update on this yet, @jrgarciadev ?

**Comment by eduardscript** [comment]
Any updates?

**Comment by mayyusan101** [comment]
Any update?

**Comment by Pr1ms** [comment]
Hello. 
Does anyone have a workaround for rendering `Table` elements into a server page (which cannot be made a client page without rewriting like 30% of my application and infringing constraints I have to enforce).
I've tried reexporting `Table `and its sub-components from within a `use client` context but it still generate the error : `Error: Unknown element <[object Object]> in collection.`

**Comment by CMarshall92** [comment]
Not only is this an issues the types are incorrect as well, its expecting react nodes and the components are returning reactElements

![image](https://github.com/nextui-org/nextui/assets/33270926/b7b3a028-a85f-4e4c-afa6-79b59df05462)

**Comment by Poweranimal** [comment]
> Hey @iyxan23 we are waiting for the next [react-aria](https://github.com/adobe/react-spectrum) release to get this issue fixed, the current collection approach is compiled by next.js in a different way so it removes the `use client;` directive, in the meantime please add the directive on top of your files for collection based components -> Table, Accordion, Dropdown, and Tabs.

@jrgarciadev 
Would you be so kind and share with us the issue of react-aria that fixes this issue?

**Comment by pablojsx** [comment]
well

**Comment by tookender** [comment]
@jrgarciadev 
Would you be so kind and share with us the issue of react-aria that fixes this issue?

**Comment by bdubtga** [comment]
Still having this issue trying to use dropdowns, any timeline on a update?

**Comment by usuarez** [comment]
> Still having this issue trying to use dropdowns, any timeline on a update?

Hi, try with the 'use client' y your component

**Comment by usuarez** [comment]
> This works but breaks SSR

Every nextUi component has the 'use client' declared at top, and i don't know if the dev want to change it, finally i decided to use primereact, is very easy

**Comment by stefnto** [comment]
> Hey @iyxan23 we are waiting for the next [react-aria](https://github.com/adobe/react-spectrum) release to get this issue fixed, the current collection approach is compiled by next.js in a different way so it removes the `use client;` directive, in the meantime please add the directive on top of your files for collection based components -> Table, Accordion, Dropdown, and Tabs.

Guys did you read the dev's reply? He said you should use "use client" for now, but internally nextjs removes the 'use client' so the component is rendered in the server...

**Comment by dragonlobster** [comment]
this bug also applies to Select. the docs don't mention `use client` at all

**Comment by Veeresh-R-G** [comment]
I am facing this issue with the table component too. 
- I have wrapped my Table component in ```use client``` making it a client component
- Using the stable version of next-ui/react

![image](https://github.com/user-attachments/assets/a6fa4056-5a53-4188-97f8-c8fa151c440c)

**Comment by wingkwong** [comment]
@Veeresh-R-G it's more like you missed `TableBody`.

**Comment by Veeresh-R-G** [comment]
> @Veeresh-R-G it's more like you missed `TableBody`.

@wingkwong I had copied the code from the docs at Next-UI. For the first time I ran the code, it did show the mentioned errors, but now its working fine. 

Thanks for the prompt reply 😀

**Comment by slimshreydy** [comment]
Any updates on this issue? <Select> appears to still be broken, even when I try exporting it in a simple component that says "use client" as the top, e.g.

```
"use client"
export {Select, SelectItem} from "@nextui-org/select"
```

Instead, I have to put `"use client"` on the page where I invoke `Select`, which is not ideal.

**Comment by yanuu1337** [comment]
If you're using the Pages Directory, you don't need to worry about this. 

Unfortunately the only "fix" for now (for App Directory projects) is to move your element (whether it's a Table, an Accordion or some other element with this issue) to a seperate Client Component and render that where you need to.

**Comment by citrusjunoss** [comment]
it's not work
```typescript
// /utils/nextui.ts
"use client";

import { Accordion, AccordionItem } from "@nextui-org/accordion";

export { Accordion, AccordionItem };
```

```tsx
// /app/veterinary/page.tsx

import { Accordion, AccordionItem } from "@/utils/nextui";
import { title } from "@/components/primitives";

const VeterinaryPage = async () => {
  const res = await fetch(
    `${process.env.API_HOST}/categories/categories?source=1`,
  );
  const data = (await res.json()) || [];

  return (
    <div className="flex flex-col items-center justify-center gap-4">
      <h1 className={title()}>专业医疗</h1>
      {/* body */}
      <div className="flex flex-wrap gap-4 mt-4">
        <Accordion variant="splitted">
          {data.map((item: any) => (
            <AccordionItem
              key={item.id}
              aria-label="Accordion 1"
              title={item.name}
            >
              {item.children.map((child: any) => (
                <span key={item.id}>{child.name}</span>
              ))}
            </AccordionItem>
          ))}
        </Accordion>
      </div>
    </div>
  );
};

export default VeterinaryPage;

```


thorow error

```
 Error: Unknown element <[object Object]> in collection.
    at $eb2240fc39a57fa5$export$bf788dd355e3a401.getFullNode (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1547:23)
    at getFullNode.next (<anonymous>)
    at $eb2240fc39a57fa5$export$bf788dd355e3a401.iterateCollection (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1503:26)
    at iterateCollection.next (<anonymous>)
    at [Symbol.iterator] (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1630:22)
    at Generator.next (<anonymous>)
    at new $05ca4cd7c4a5a999$export$863faf230ee2118a (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:841:18)
    at /Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1729:454
    at /Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1679:16
    at $7613b1592d41b092$export$6cd28814d92fa9c9 (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1673:241)
    at $875d6693e12af071$export$728d6ba534403756 (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:1729:209)
    at useAccordion (/Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:2776:203)
    at /Volumes/PersonalDisk/GitHub/ahri-web/.next/server/chunks/ssr/node_modules_49bccb._.js:3107:362
```

**Comment by yanuu1337** [comment]
@citrusjunoss You can't use these elements with server components (for now I guess), also, why would you import the Accordion in a CC file and export it? 

Just make the VeterinaryPage not async, use the "use client" directive, wrap the fetch in a useEffect, store the data as state and you're good. Take a look at these docs for [fetching](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching)

**Comment by citrusjunoss** [comment]
> @citrusjunoss You can't use these elements with server components (for now I guess), also, why would you import the Accordion in a CC file and export it?
> 
> Just make the VeterinaryPage not async, use the "use client" directive, wrap the fetch in a useEffect, store the data as state and you're good. Take a look at these docs for [fetching](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching)

Do I have to give up server-side rendering for my entire page because of this component, if so, then give up on this component, server-side rendering is more important to me

BTW, I see the official website describes it this way, won't he take effect if I put it in a file like this?

<img width="988" alt="image" src="https://github.com/user-attachments/assets/ba21a27f-188b-4804-9776-e14e98ea492b">

**Comment by yanuu1337** [comment]
@citrusjunoss Yes, you can export part of your code into a new file and use it in your server component (as long as the new file has a "use client" directive)

**Comment by jjangga0214** [comment]
It's been a while. Shouldn't this be documented as note?

**Comment by KUN1007** [comment]
Among these components that pose challenges for SSR, the most impactful one is the `Tabs` component.
For this component, if the `defaultSelectedKey` property is set, SSR will function (though there will be unsightly hydration errors in the console, but if you don’t mind that, everything works as expected).
On this basis, when the page initially renders, the selected item on the Tabs will flicker, causing visual disruption for users. Therefore, we can set `disableAnimation={true}` and change it to `disableAnimation={false}` once hydration is complete, achieving a simple, usable `Tabs` component.
Unfortunately, there will still be hydration errors in the console :(

**Comment by Jimflip** [comment]
I'm confused, there have been 9 react-aria releases since this issue was raised, have any of them addressed this problem?

If I understand correctly many components are only client side due to the use of 'use client', is that correct?

**Comment by nthsense** [comment]
I'm guessing react-aria is never going to fix this. The reasoning is here: https://github.com/adobe/react-spectrum/pull/5826

**Comment by FTCHD** [comment]
add `Breadcrumbs` to the list of components that don't work unless you have `use client` at the top. 

**this should be specified in the Docs**, people could spend hours debugging this in Next thinking it's something they did wrong.

**Comment by FTCHD** [comment]
> > If anyone else comes here, this is what it's supposed to look like, and it does fix this error perfectly:
> > ![Screenshot from 2023-09-27 17-03-02](https://github.com/nextui-org/nextui/assets/595049/b44266da-ae29-4aa6-857e-94c78f994e01)
> 
> No need to mark the whole page or larger server components as client components because of this.
> 
> You can just do this
> 
> ```
> "use client"
> export { Select } from "@nextui-org/select"
> ```
> 
> Now you can just import the component from this file and you're done

this is not a solid solution, it works for some components but not for others (like `Breadcrumbs`, both from `/react` or `/breadcrumbs` import). it's also extremely counterintuitive to do this when using a UI library supposedly made for Next.js

**Comment by LubomirGeorgiev** [comment]
Any update here?

**Comment by Scholar01** [comment]
I guess the name was changed to heroui because it will never support nextjs? So calling it nextui is like a joke?

**Comment by theromanticsoul** [comment]
+1

**Comment by tiaaago** [comment]
Any updates?

**Comment by wingkwong** [comment]
Closing in favor of v3. If the issue is reproducible in v3, please create a new one with a minimal reproducible example.
